systems/plants/RigidBodyManipulator.cpp

```diff
@@ -533,6 +534,12 @@ void RigidBodyManipulator::compile(void)
   cached_q.resize(num_positions);
   cached_v.resize(num_velocities);
 
+  cached_q_new.resize(num_positions);
+  cached_v_new.resize(num_velocities);
+
+  cached_q_old.resize(num_positions);
+  cached_v_old.resize(num_velocities);
```

Comment:
Had to separate these so that I know specifically whether the old doKinematics and/or new doKinematics has been cached with a given q and v. There was a place outside of RigidBodyManipulator that used cached_q and should continue to work for both new and old doKinematics, so I kept cached_q and cached_v as well (set by whatever doKinematics method was called last).